dandi/dandiapi.py

```diff
@@ -318,10 +318,16 @@ def __init__(
         self, api_url: Optional[str] = None, token: Optional[str] = None
     ) -> None:
         """
-        Construct a client instance from a server's base API URL and an
-        optional authentication token/API key.  If no URL is supplied, the URL
-        of the instance named in the :envvar:`DANDI_INSTANCE` environment
-        variable (default value: ``dandi``) is used.
+        Construct a client instance.
+
+        :param api_url: API URL. For DANDI production, use ``"dandi"``. For DANDI staging,
```

Comment:
*For DANDI production, use ``"dandi"``* — This is wrong.  For production, use `"https://api.dandiarchive.org/api"` or pass `"dandi"` to the `for_dandi_instance()` classmethod.

Reply:
@jwodder oh, I see. How would you feel about adding:

```
if api_url in known_instances:
    api_url = known_instances[api_url].api
```

then the user can provide "dandi" , dandi-api-local-docker-tests, and "dandi-staging". I could see why you might not want this, since these are not "URL"s, but I feel like it is more user friendly, particularly if the API addresses change down the road. Maybe we could change the variable name from `api_url` to `api` so that it can accept the instance names without being mislabeled. My first preference would be to do this, which would be changing the API to the way I intuitively expected it to work, but if you don't think that's a good idea I think improving the docstring would be sufficient.

Reply:
No.  There's already a way to construct clients from instance names, and it's the `for_dandi_instance()` classmethod.